tests/test_basic.py

```diff
@@ -240,13 +240,15 @@ def test_add_storage(rook_cluster):
 
     # wait for the additional osd
     # this may take a while
-    while osds_expected != osds_new:
+    while osds == osds_new:
```

Comment:
This is okay only because we're only waiting on 1 OSD. So this loop will finish as soon as the number of OSDs has changed. In most cases this is because the new OSD is ready. However, if we were waiting to go from 3 to 5 OSDs, then the `get_number_of_osds()` may return 4, which will end the loop before the last OSD is ready.

Reply:
You're right, interesting that we have vise-versa situation when we had 3 and expect 4 but have 5 OSDs, because the node from previous failed test is up during this wait loop.